### PR TITLE
T10 completed

### PR DESCRIPTION
**Pull request by YoganathanM**

### CODE REVIEW

File: Assignment/Program.cs [hunk 1]
```diff
@@ -1,7 +1,19 @@
-﻿namespace Assignment {
-   internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿int reverse = 0;
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:1]: This can be inside the while loop just after int number = originalNumber;

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]
1. If the input number is 1240 it is not reversed properly.
2. Again there seems to be some code in main for this branch which is unnecessary.
3. Check .gitignore. It is not handling .vs bin and obj folders properly.

**Comment by varalakshmi-imm** [comment]
@YoganathanM Seems 1240 has to return 421 and for 50 the reverse is 5 and your implementation works as expected.
I assumed that it should be 0421 and should work for negative numbers too. So push the other changes.

File: Assignment/Program.cs [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿int reverse = 0;
+while (true) {
+   Console.Write ("Enter number :");
+   if (int.TryParse (Console.ReadLine (), out int originalNumber) && originalNumber > 0) {
+      int number = originalNumber;
+      while (number > 0) {
+         int remainder = number % 10;
+         reverse = reverse * 10 + remainder;
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:8]: Could have avoided the variable remainder and used the computation directly on reverse.

File: Assignment/Program.cs [hunk 3]
```diff
@@ -1,7 +1,19 @@
...
+         number = number / 10;
```
Comment (varalakshmi-imm) [inline Assignment/Program.cs:9]:
```suggestion
         number /= 10;

```

**Review by varalakshmi-imm — CHANGES_REQUESTED** [review]

File: Assignment/Program.cs [hunk 4]
```diff
@@ -1,7 +1,19 @@
-﻿namespace Assignment {
-   internal class Program {
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello world!");
+﻿int reverse = 0;
+while (true) {
+   Console.Write ("Enter number :");
+   if (int.TryParse (Console.ReadLine (), out int originalNumber) && originalNumber > 0) {
+      int number = originalNumber;
+      while (number > 0) {
```
[the remark this answers is not in the document]
Comment (YoganathanM) [reply Assignment/Program.cs:6]: Handling invalid input.

File: Assignment/Program.cs (hunk 2 above)
Comment (YoganathanM) [reply Assignment/Program.cs:8]: ok

File: Assignment/Program.cs (hunk 1 above)
Comment (YoganathanM) [reply Assignment/Program.cs:1]: ok

File: Assignment/Program.cs (hunk 3 above)
Comment (YoganathanM) [reply Assignment/Program.cs:9]: ok

**Review by varalakshmi-imm — APPROVED** [review]
@YoganathanM This implementation doesn't handle numbers that end with zero and negative numbers.
As an extended exercise do handle it later.

**Pull request closed without merging**